mod-source-record-manager-server/src/main/java/org/folio/services/AbstractChunkProcessingService.java

```diff
@@ -56,6 +74,71 @@ public Future<Boolean> processChunk(RawRecordsDto incomingChunk, String jobExecu
+    if (!CollectionUtils.isEmpty(childProfiles)) {
+      isExists = isExistsMatchProfileToInstanceWithActionUpdateMarcBib(childProfiles);
+    }
+    return isExists;
+  }
+
+  private boolean isExistsMatchProfileToInstanceWithActionUpdateMarcBib(Collection<ProfileSnapshotWrapper> profileSnapshotWrapperList) {
+    for (ProfileSnapshotWrapper profileSnapshotWrapper : profileSnapshotWrapperList) {
+      if ((profileSnapshotWrapper.getContentType() == MATCH_PROFILE) && (isMatchingMarcBibToInstance(profileSnapshotWrapper))) {
+        ProfileSnapshotWrapper actionProfileSnapshotWrapper = getChildSnapshotWrapperByType(profileSnapshotWrapper, ACTION_PROFILE);
+        if (actionProfileSnapshotWrapper != null) {
+          ProfileSnapshotWrapper mappingProfileSnapshotWrapper = getChildSnapshotWrapperByType(actionProfileSnapshotWrapper, MAPPING_PROFILE);
+          if ((mappingProfileSnapshotWrapper != null) && (isMappingMarcBibToMarcBib(mappingProfileSnapshotWrapper))) {
+            return true;
+          }
+        }
+        isExistsMatchProfileToInstanceWithActionUpdateMarcBib(profileSnapshotWrapper.getChildSnapshotWrappers());
+      }
+    }
+    return false;
+  }
+
+  private ProfileSnapshotWrapper getChildSnapshotWrapperByType(ProfileSnapshotWrapper profileSnapshotWrapper,
+                                                                       ProfileSnapshotWrapper.ContentType contentType) {
+    if (profileSnapshotWrapper.getChildSnapshotWrappers() != null && !profileSnapshotWrapper.getChildSnapshotWrappers().isEmpty()) {
+      List<ProfileSnapshotWrapper> childSnapshotWrappers = profileSnapshotWrapper.getChildSnapshotWrappers();
+      for(ProfileSnapshotWrapper snapshotWrapper : childSnapshotWrappers) {
+        if (snapshotWrapper.getContentType() == contentType) {
+          return snapshotWrapper;
+        }
+      }
+    }
+    return null;
+  }
+
+  private boolean isMatchingMarcBibToInstance(ProfileSnapshotWrapper profileSnapshotWrapper) {
+    boolean result = false;
+    MatchProfile matchProfile = extractProfile(profileSnapshotWrapper, MatchProfile.class);
+    if (!CollectionUtils.isEmpty(matchProfile.getMatchDetails())) {
+      MatchDetail matchDetail = matchProfile.getMatchDetails().get(0);
+      result = matchDetail.getIncomingRecordType() == EntityType.MARC_BIBLIOGRAPHIC
+        && matchDetail.getExistingRecordType() == EntityType.INSTANCE;
+    }
+    return result;
+  }
+
+  private boolean isMappingMarcBibToMarcBib(ProfileSnapshotWrapper profileSnapshotWrapper) {
+    MappingProfile mappingProfile = extractProfile(profileSnapshotWrapper, MappingProfile.class);
+    return mappingProfile.getIncomingRecordType() == EntityType.MARC_BIBLIOGRAPHIC
+      && mappingProfile.getExistingRecordType() == EntityType.MARC_BIBLIOGRAPHIC;
+  }
+
+  private <T> T extractProfile(ProfileSnapshotWrapper profileSnapshotWrapper, Class clazz) {
```

Comment:
Is there any exception handling should this method fail? class cast or json decode exceptions.

Reply:
This exception will be handled at a higher level.
Example:
![image](https://user-images.githubusercontent.com/8954252/183622302-7a17cdb4-fa88-4166-8ca9-d43271b8e5af.png)